compiler/semtypes.nim

```diff
@@ -1153,12 +1153,14 @@ proc semProcTypeNode(c: PContext, n, genericParams: PNode,
     # turn explicit 'void' return type into 'nil' because the rest of the
     # compiler only checks for 'nil':
     if skipTypes(r, {tyGenericInst, tyAlias, tySink}).kind != tyVoid:
+      if kind == skProc and r.kind in {tyStmt, tyExpr}:
```

Comment:
What about ``skFunc, skMethod, skIterator, skConverter``? I would stick to ``kind notin {skMacro, skTemplate}``

Reply:
Done